### DISCUSSION TITLE
Using Nuxt 3 plugin on a server API route

### DISCUSSION THREAD

**Discussion by charbs-io** [nuxt/nuxt#22386] [Questions]
Hi all,

I've got a Nuxt 3 plugin that I am trying to run on server API routes, but having some issues.

```
// /plugins/pocketbase.ts

const pocketbase...

// access to the pocketbase instance and methods...

export default defineNuxtPlugin(() => {
  return {
    provide: {
      pocketbase,
    },
  };
});
```

I'm consuming this inside a vue page, with no issues at all.
```
// /pages/login.vue

<script lang="ts" setup>
const { $pocketbase } = useNuxtApp();

function login() {
  // I have access to $pocketbase methods etc 👍
  $pocketbase...
}
</script>
```

⚠️ **But Nuxt is throwing a 500 (useNuxtApp is not defined) when I try and use this same code on a server API route.**

```
// /server/api/auth/login.post.ts

export default defineEventHandler(async (event) => {
  const { $pocketbase } = useNuxtApp();

  // rest of code
});
```

I'm very new to Nuxt (have been using Vue 3 for many years though) - so any help would be great!

For additional context:

Pocketbase is a Go-based backend, comes with a JS SDK

Instead of interfacing directly with the Pocketbase SDK, I wanted to have this logic in server routes in Nuxt; this is to enable me to do certain things like send emails, logs, notifications etc as the user invokes certain routes

The vue components and pages would interface with the Nuxt API server routes

The approach to the architecture may be wrong, it's okay, this is for a hobby project and i'm trying to find what's possible with Nuxt 3

Thanks all

**Comment by xjccc** [comment]
![image](https://github.com/nuxt/nuxt/assets/16478776/144b8983-719b-4c7c-af22-2d68a6c95c09)
is it right?
or perhaps have another way?

**Comment by huang-julien** [comment]
A server API route doesn't initialize any Vue app to render so plugins won't be run
